File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
         int start = 0;
        int last = s.length() - 1;
        while (start <= last) {
            if (!Character.isLetterOrDigit(s.charAt(start))) {
                start++;
            } else if (!Character.isLetterOrDigit(s.charAt(last))) {
                last--;
            } else if (Character.toLowerCase(s.charAt(start)) != Character.toLowerCase(s.charAt(last))) {
                return false;
            } else {
                start++;
                last--;
            }
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int start = 0;
        int last = s.size() - 1;
        while (start <= last)
        {
            if (!isalnum(s[start]))
            {
                start++;
            }
            else if (!isalnum(s[last]))
            {
                last--;
            }
            else if (tolower(s[start]) != tolower(s[last]))
            {
                return false;
            }
            else
            {
                start++;
                last--;
            }
        }
        return true;
    }
};